buildSrc/src/main/kotlin/org/gradle/plugins/buildscan/BuildScanConfigurationPlugin.kt

```diff
@@ -120,81 +127,79 @@ class BuildScanConfigurationPlugin : Plugin<Project> {
             }
         }
 
-        execAsync(Runnable {
+        execAsync {
             val commitId = run("git", "rev-parse", "--verify", "HEAD")
             setCommitId(commitId)
             val status = run("git", "status", "--porcelain")
-            if (!status.isEmpty()) {
+            if (status.isNotEmpty()) {
                 buildScan {
                     tag("dirty")
                     value("Git Status", status)
                 }
             }
-        })
+        }
 
-        execAsync(Runnable {
+        execAsync {
             val branchName = run("git", "rev-parse", "--abbrev-ref", "HEAD")
-            if (!branchName.isEmpty() && branchName != "HEAD") {
-                buildScan.tag(branchName)
-                buildScan.value("Git Branch Name", branchName)
+            if (branchName.isNotEmpty() && branchName != "HEAD") {
+                buildScan {
+                    tag(branchName)
+                    value("Git Branch Name", branchName)
+                }
             }
-        })
+        }
     }
 
-    private fun Project.extractBuildCacheData() {
+    private
+    fun Project.extractBuildCacheData() {
         if (gradle.startParameter.isBuildCacheEnabled) {
             buildScan.tag("CACHED")
 
-            val tasksToInvestigate = System.getProperty("cache.investigate.tasks")?.split(",") ?: listOf(":baseServices:classpathManifest")
+            val tasksToInvestigate = System.getProperty("cache.investigate.tasks", ":baseServices:classpathManifest")
+                .split(",")
 
-            project.buildScan.buildFinished({
-                allprojects.flatMap { it.tasks }.forEach {
-                    if (it.state.executed && (it.path in tasksToInvestigate)) {
+            buildScan.buildFinished {
+                allprojects.flatMap { it.tasks }
+                    .filter { it.state.executed && it.path in tasksToInvestigate }
+                    .forEach { task ->
                         val hasher = (gradle as GradleInternal).services.get(ClassLoaderHierarchyHasher::class.java)
-                        Visitor(project.buildScan, hasher, it).visit(it::class.java.classLoader)
+                        Visitor(buildScan, hasher, task).visit(task::class.java.classLoader)
                     }
-                }
-            })
+            }
         }
     }
 
-    private fun Project.extractAllReportsFromCI(project: Project) {
+    private
+    fun Project.extractAllReportsFromCI() {
         val capturedReportingTypes = listOf("html") // can add xml, text, junitXml if wanted
         val basePath = "${System.getenv("BUILD_SERVER_URL")}/repository/download/${System.getenv("BUILD_TYPE_ID")}/${System.getenv("BUILD_ID")}:id"
 
         gradle.taskGraph.afterTask {
-            if (state.failure != null) {
-                if (this is Reporting<*>) {
-                    val reportContainer = this as ReportContainer<*>
```

Comment:
Note that this was casting the task but it should use `task.reports` instead, see below